src/assets/js/fullpostpage/unlock-content/edit-fullpp.js

```diff
@@ -0,0 +1,302 @@
+
+
+  //get current post id
+  const getpost = useSelect((select) => {
+    const { getCurrentPostId } = select("core/editor");
+    const id = getCurrentPostId();
+  
+    return {
+      id,
+    };
+  }, []);
+
+  //check if current post is already 
+  const { isSavingPost, didPostSaveRequestSucceed } = useSelect((select) => {
+    const { isSavingPost, didPostSaveRequestSucceed } = select("core/editor");
+    return {
+      isSavingPost: isSavingPost(),
+      didPostSaveRequestSucceed: didPostSaveRequestSucceed(),
+    };
+  }, []);
+
+
+  //fetch locks attributes
+  const fetchLockSettings = async (post_id) => {
+    try {
+      const response = await apiFetch({
+        path: `/unlock-protocol/v1/get_unlockpfullpp_attributes?post_id=${post_id}`,
+      });
+
+      if (response && response.locks) {
+        onUpdateLocks(response.locks);
+      }
+    } catch (error) {
+      console.error("Error fetching lock settings:", error);
+    }
+  };
+
+  //useEffect 1: that fetches the networks
+  useEffect(() => {
+    apiFetch({
+      path: "/unlock-protocol/v1/settings",
+    })
+      .then((resp) => {
+        let networks = resp.networks;
+        let selectOptions = [
+          {
+            label: __("None", "unlock-protocol"),
+            value: -1,
+          },
+        ];
+
+        Object.entries(networks).forEach(([key, item]) => {
+          selectOptions.push({
+            label: item.network_name,
+            value: item.network_id,
+          });
+        });
+
+        onUpdateEthereumNetworks(selectOptions);
+
+        // Fetch lock settings from the post meta after networks are fetched
+        fetchLockSettings(getpost.id);
+      })
+      .catch((err) => { });
+  }, [getpost.id, refresh]); // refresh the useEffect anytime either of this changes
+
+
+  //useEffect 2: called when post/page publish/update button clicked
+  useEffect(() => {
+    if (!isSavingPost && didPostSaveRequestSucceed) {
+
+      // refresh the useEffect 1 to reload meta panel in post/page editor sidebar when publish/update button is clicked
+      setRefresh(!refresh); 
+    }
+  }, [isSavingPost, didPostSaveRequestSucceed, getpost.id]);
+
+
+  // Check if the Ethereum address is valid
+  const isAddressValid = (address) => {
+    return /^0x[a-fA-F0-9]{40}$/.test(address);
+  };
+
+  // Save the attributes object to the post meta
+  const saveAttributes = async (locks, post_id) => {
+
+    // Filter the locks to only include those with a valid address and network
+    const validLocks = locks.filter(
+      (lock) => lock.network !== -1 && lock.address && isAddressValid(lock.address)
+    );
+
+    if (validLocks.length === 0) {
+      setSaveMessage("No valid lock(s) attributes to save.");
+      return;
+    }
+
+    const attributes = {
+      locks: validLocks,
+      ethereumNetworks: ethereumNetworks,
+    };
+
+    try {
+      const response = await apiFetch({
+        path: "/unlock-protocol/v1/save_unlockpfullpp_attributes",
```

Comment:
Plese update the file name  for readability and consistency!

Reply:
name updated. fixed